external-builds/pytorch/build_prod_wheels.py

```diff
@@ -605,47 +605,48 @@ def do_build_pytorch(
-            env["USE_FBGEMM_GENAI"] = "ON"
-            print(
-                f"FBGEMM_GENAI enabled (PyTorch < 2.10, Linux): {env['USE_FBGEMM_GENAI'] == 'ON'}"
-            )
+        if args.enable_pytorch_fbgemm_genai_linux is None:
+            # Default behavior — based on PyTorch version
+            if pytorch_build_version_parsed.release < (2, 8):
+                use_fbgemm_genai = "ON"
+            else:
+                use_fbgemm_genai = "OFF"
+            print(f"FBGEMM_GENAI default behavior based on version: {use_fbgemm_genai}")
         else:
-            env["USE_FBGEMM_GENAI"] = (
-                "ON" if args.enable_pytorch_fbgemm_genai_linux else "OFF"
-            )
+            # Explicit override: user has set the flag to true/false
+            use_fbgemm_genai = "ON" if args.enable_pytorch_fbgemm_genai_linux else "OFF"
+            print(f"FBGEMM_GENAI override set by flag: {use_fbgemm_genai}")
+
+        env["USE_FBGEMM_GENAI"] = use_fbgemm_genai
+        print(f"FBGEMM_GENAI enabled: {env['USE_FBGEMM_GENAI'] == 'ON'}")
+
+        if args.enable_pytorch_flash_attention_linux is None:
+            # Default behavior — determined by PyTorch version
+            if pytorch_build_version_parsed.release < (2, 8):
+                use_flash_attention = "1"
+            else:
+                use_flash_attention = "0"
             print(
-                f"FBGEMM_GENAI enabled (PyTorch >= 2.10, Linux): {env['USE_FBGEMM_GENAI'] == 'ON'}"
-            )
-
-        # Enabling/Disabling Flash attention based on Pytorch version in Linux
-        if pytorch_build_version_parsed.release < (2, 10):
-            env.update(
-                {
-                    "USE_FLASH_ATTENTION": "1",
-                    "USE_MEM_EFF_ATTENTION": "1",
-                }
-            )
-            print(
-                f"Flash Attention enabled (PyTorch < 2.10, Linux): {env['USE_FLASH_ATTENTION'] == '1'}"
+                f"Flash Attention default behavior based on version: {use_flash_attention}"
             )
         else:
+            # Explicit override: user has set the flag to true/false
             use_flash_attention = (
                 "1" if args.enable_pytorch_flash_attention_linux else "0"
             )
-            env.update(
-                {
-                    "USE_FLASH_ATTENTION": use_flash_attention,
-                    "USE_MEM_EFF_ATTENTION": use_flash_attention,
-                }
-            )
-            print(
-                f"Flash Attention enabled (PyTorch >= 2.10, Linux): {env['USE_FLASH_ATTENTION'] == '1'}"
-            )
+            print(f"Flash Attention override set by flag: {use_flash_attention}")
+
+        env.update(
+            {
+                "USE_FLASH_ATTENTION": use_flash_attention,
+                "USE_MEM_EFF_ATTENTION": use_flash_attention,
+            }
+        )
+        print(f"Flash Attention enabled: {env['USE_FLASH_ATTENTION'] == '1'}")
```

Comment:
do we want to add that we do not on;ly enable flash attention but also memory efficiency attention?